System/System.cpp

```diff
@@ -45,16 +48,27 @@ namespace RTE {
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	bool System::PathExistsCaseSensitive(const std::string &pathToCheck) {
-#ifndef __linux__
+		#ifndef __linux__
 		if (s_CaseSensitive) {
 			if (s_WorkingTree.empty()) {
-				for (const std::filesystem::directory_entry &directoryEntry : std::filesystem::recursive_directory_iterator(s_WorkingDirectory)) {
+				for (const std::filesystem::directory_entry &directoryEntry : std::filesystem::recursive_directory_iterator(s_WorkingDirectory, std::filesystem::directory_options::follow_directory_symlink)) {
 					s_WorkingTree.emplace_back(std::hash<std::string>()(directoryEntry.path().generic_string().substr(s_WorkingDirectory.length())));
 				}
+				std::cout << "built dir_iterator with size: " << s_WorkingTree.size() << std::endl;
+			}
+			bool exists = std::find(s_WorkingTree.begin(), s_WorkingTree.end(), std::hash<std::string>()(pathToCheck)) != s_WorkingTree.end();
+			if (exists) {
```

Comment:
You can probably inline exists into the if check, doesn't seem to be used anywhere else